lib/fortnox/api/models/base.rb

```diff
@@ -19,8 +19,12 @@ def self.attribute( name, *args )
         end
 
         def self.new( hash )
-          obj = preserve_meta_properties( hash ) do
-            super( hash )
+          begin
+            obj = preserve_meta_properties( hash ) do
+              super( hash )
+            end
+          rescue Dry::Struct::Error, Dry::Types::ConstraintError, Dry::Types::MissingKeyError => e
```

Comment:
We should catch these independently and preserve the information. That means adding two additional exception types and changing this to:

```
rescue Dry::Struct::Error => e
  raise Fortnox::API::AttributeError.new e
rescue Dry::Types::ConstraintError => e
  raise Fortnox::API::InvalidAttributeValueError.new e
rescue Dry::Types::MissingKeyError => e
  raise Fortnox::API::MissingAttributeError.new e
end
```